File: backend/internal/models/favour.go
package models

import "time"

type FavourChoice struct {
	ID          string    `json:"id"`
	Label       string    `json:"label"`
	Description *string   `json:"description"`
	Costs       int       `json:"costs"`
	CanMessage  bool      `json:"can_message"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

type FavourCount struct {
	Total     int `json:"total"`
	Remaining int `json:"remaining"`
}

type FavourRequest struct {
	ID                string     `json:"id"`
	ChoiceID          string     `json:"choice_id"`
	ChoiceLabel       string     `json:"choice_label"`
	ChoiceDescription *string    `json:"choice_description"`
	Message           *string    `json:"message"`
	RequestedAt       time.Time  `json:"requested_at"`
	FulfilledAt       *time.Time `json:"fulfilled_at"`
}

File: client/src/lib/types/favour.ts
export interface FavourChoice {
	id: string;
	label: string;
	description: string | null;
	cost: number;
	can_message: boolean;
	created_at: string;
	updated_at: string;
}

export interface FavourCount {
	total: number;
	remaining: number;
}

export interface FavourRequest {
	id: string;
	choice_id: string;
	choice_label: string;
	choice_description: string | null;
	message: string | null;
	requested_at: string;
	fulfilled_at: string | null;
}
